### PR TITLE
log warning if unsafe use of RPC and account unlocking

### PR DESCRIPTION
**Pull request by whilei**
Yields:

```
$ geth --chain morden --unlock 0 --rpc
2018-05-21 13:55:27 Geth Classic version: whilei.v5.3.0+2-b4b0c9f[branch=feat/unlock-rpc-warn,commit=problem:-unsafe-usage-is-accompanied-with-a-usage-warning,built=May/21Mon-13:55:19]
2018-05-21 13:55:27 Blockchain: Morden Testnet
2018-05-21 13:55:27 Chain database: /Users/ia/Library/EthereumClassic/morden/chaindata
2018-05-21 13:55:27 Blockchain upgrades configured: 6
2018-05-21 13:55:27   494000 Homestead
2018-05-21 13:55:27  1783000 GasReprice
2018-05-21 13:55:27  1885000 The DAO Hard Fork
2018-05-21 13:55:27  1915000 Diehard
2018-05-21 13:55:27  2000000 Gotham
2018-05-21 13:55:27  2300000 Defuse Difficulty Bomb
2018-05-21 13:55:27 --------------------------------------------------------------------------------------------------------------
2018-05-21 13:55:27 *
2018-05-21 13:55:27

WARNING: Unsafe use of --unlock and exposed RPC API [ currently: eth,net,web3 ].
It's unsafe to unlock an account while exposing ANY of the following RPC APIs: [eth personal admin]
You can use the flag --rpc-api to enable only certain RPC API modules if necessary.


2018-05-21 13:55:27 *
2018-05-21 13:55:27 --------------------------------------------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by splix** [comment]
Let's add more information. It's not clear what "unsafe" means. It should be clearly stated that anyone from the internet will be able to transfer funds from unlocked account without any password.

**Comment by whilei** [comment]
```
2018-05-22 06:12:58  1783000 GasReprice
2018-05-22 06:12:58  1885000 The DAO Hard Fork
2018-05-22 06:12:58  1915000 Diehard
2018-05-22 06:12:58  2000000 Gotham
2018-05-22 06:12:58  2300000 Defuse Difficulty Bomb
2018-05-22 06:12:58 --------------------------------------------------------------------------------------------------------------
2018-05-22 06:12:58 *
2018-05-22 06:12:58

> WARNING: Unsafe use of --unlock and exposed RPC API [ currently: eth,net,web3 ].
> It's UNSAFE to unlock an account while exposing ANY of the following RPC APIs: [eth personal admin]
> 	Anyone from the internet will be able to transfer funds from an unlocked account without any password.
> You can use the --rpc-api flag to enable only certain RPC API modules if necessary.


2018-05-22 06:12:58 *
2018-05-22 06:12:58 --------------------------------------------------------------------------------------------------------------
```

**Comment by splix** [comment]
It seems it will display a message even if there will be no real danger, like listening on 127.0.0.1. It should allow to listen on 127.0.0.1/8, 192.168.0.1/32 and 10.0.0.1/8 without such warning, but display it if it listen on all interfaces or non-local one.

**Comment by whilei** [comment]
Added 
- conditional for internet-facing listen addresses and additional information in the warning text about this
- prompt user to continue in spite of warning

```
$ geth --chain morden --rpc --unlock 0 -rpc-addr 193.168.0.1
2018-05-23 17:08:20 Geth Classic version: whilei.v5.3.0+6-7d055ef[branch=feat/unlock-rpc-warn,commit=problem:-localhost-not-parsed,-so-default-rpc-local-got-false-pos-warn,built=May/23Wed-17:07:39]
2018-05-23 17:08:20 Blockchain: Morden Testnet
2018-05-23 17:08:20 Chain database: /Users/ia/Library/EthereumClassic/morden/chaindata
2018-05-23 17:08:20 Blockchain upgrades configured: 6
2018-05-23 17:08:20   494000 Homestead
2018-05-23 17:08:20  1783000 GasReprice
2018-05-23 17:08:20  1885000 The DAO Hard Fork
2018-05-23 17:08:20  1915000 Diehard
2018-05-23 17:08:20  2000000 Gotham
2018-05-23 17:08:20  2300000 Defuse Difficulty Bomb
2018-05-23 17:08:20 --------------------------------------------------------------------------------------------------------------
2018-05-23 17:08:20 *
2018-05-23 17:08:20

>    !!!  WARNING: Unsafe use of --unlock and exposed RPC API [ currently: eth,net,web3 ].  !!!
>
>    It's UNSAFE to unlock an account while exposing ANY of the following RPC APIs: [eth personal admin] to the internet.
>    Anyone from the internet will be able to transfer funds from an unlocked account without any password.
>
>    You can use the --rpc-api flag to enable specific RPC API modules if necessary,
>    and/or restrict the exposed RPC listen address with the --rpc-addr flag.


2018-05-23 17:08:20 *
2018-05-23 17:08:20 --------------------------------------------------------------------------------------------------------------
2018-05-23 17:08:20 Do you really wish to proceed? [y/n]:
```

**Comment by whilei** [comment]
@splix ⬆️ LGTY?

**Comment by splix** [comment]
yes, LGTM

**Comment by whilei** [comment]
@splix please an official review approval, then i'll get this in

File: cmd/geth/log_context.go [hunk 1]
```diff
@@ -366,6 +366,9 @@ func logIfUnsafeConfiguration(ctx *cli.Context) {
 		if configuredRPCListenAddr == "*" {
 			return true
 		}
+		if strings.Contains(configuredRPCListenAddr, "localhost") {
```
[the remark this answers is not in the document]
Comment (whilei) [reply cmd/geth/log_context.go]: Yes, loopback and other local interfaces are [whitelisted](https://github.com/ethereumproject/go-ethereum/pull/600/files#diff-e98cb937cfbb55774b9a4c67aadfa19bR345).

**Pull request merged**